Automattic/WP-Job-Manager
# Fix dependencies (npm/composer) problems

### Changes proposed in this Pull Request

* Update NPM Lock file to v2;
* Remove/update some old dependencies that limited the project to be installed only with Composer 1;
* Add and configure Husky as a replacement for `bjornjohansen/wp-pre-commit-hook` (which was removed by this PR as it's not updated anymore and doesn't support Composer v2);
* Also update NPM minimum version to 7.0;

### Testing instructions

* Download the repository in some new folder;
* Run `composer install` using Composer v2 or newer;
* Run `npm install` using NPM v7 or newer;
* If possible, change one of the PHP files without respecting any style guides and try to commit it to check if the pre-commit hook correctly catches the change;
